5-app-infra/modules/env_base/main.tf

```diff
@@ -16,32 +16,54 @@
 
 locals {
   environment_code = element(split("", var.environment), 0)
+  env_project_ids = {
+    "sample-base"     = data.terraform_remote_state.projects_env.outputs.base_shared_vpc_project,
+    "sample-floating" = data.terraform_remote_state.projects_env.outputs.floating_project,
+    "sample-peering"  = data.terraform_remote_state.projects_env.outputs.peering_project,
+    "sample-restrict" = data.terraform_remote_state.projects_env.outputs.restricted_shared_vpc_project,
+  }
+  env_project_id        = local.env_project_ids[var.project_suffix]
+  subnetwork_self_links = data.terraform_remote_state.projects_env.outputs.base_subnets_self_links
+  subnetwork_self_link  = [for subnet in local.subnetwork_self_links : subnet if length(regexall("regions/${var.region}/subnetworks", subnet)) > 0][0]
```

Comment:
```hcl
length(regexall("regions/${var.region}/subnetworks", subnet)) > 0
```
is checking if the current **subnet** been validated in the `for` loop is from `var.region`. 
if true it will be `> 0`
if false it will be `= 0`

```sh
~$ terraform console
> length(regexall("regions/us-central1/subnetworks", "regions/us-central1/subnetworks/subnet1"))
1
> length(regexall("regions/us-central1/subnetworks", "regions/us-east1/subnetworks/subnet2"))
0
>
```
and since there is only one subnet for each region we can use the `[0]` at the end to get the only element on the list.